src/audio/mux/mux.c

```diff
@@ -173,7 +173,8 @@ static struct comp_dev *mux_new(const struct comp_driver *drv,
 		return NULL;
 	dev->ipc_config = *config;
 
-	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+	/* allocate quite a big buffer, use rballoc to make sure it will succeed */
+	cd = rballoc(0, SOF_MEM_CAPS_RAM,
```

Comment:
I don't see why we special-case a component? something's fishy here. why not the mixer as well then?

Reply:
> I don't see why we special-case a component? something's fishy here. why not the mixer as well then?

The size of 'cd' here is > 1KB, we have very limited runtime blocks on APL, so any ask for allocation size >= 1KB are suggested to use the rballoc() and allocation will happen on the BUFFER zone.

The mixer 'cd' size is very small, using rzalloc() to allocate e.g. a 64Bytes buffer could help us to save memory usage.

Reply:
Our allocator is simple and showing its limitations. The Zephyr allocator is far better.